File: javascript/node/wesley-safadao.js
'use strict';

module.exports = (day, month, year) => {
  let safadeza, anjo, sum = 0;
  for(; month--;) sum += month;
  safadeza = sum + (year / 100) * (50 - day);
  anjo = 100 - safadeza;
  return `Você é ${safadeza.toFixed(2)}% safado e ${anjo.toFixed(2)}% anjo`;
};

File: fsharp/wesleySafadao.fs
module AsPorteirasDoSafadao 

let rec sum x = function
  | [] -> x
  | hd :: nl -> sum (hd + x) nl

let wsCalculos day month year = 
  let _naughty = (sum 0 [1..month] |> double) + ((year|>double) / 100.0) * (50.0 - (day|>double))
  let _angel = 100.0 - _naughty
  
  _naughty, _naughty

let safadeza tuple =
  let (x,y) = tuple
  sprintf "Você eh %.2f%% anjo, mas tem aquele %.2f%% de vagabundo!" x y
